flink-cdc-connect/flink-cdc-source-connectors/flink-connector-mysql-cdc/src/main/java/org/apache/flink/cdc/connectors/mysql/debezium/DebeziumUtils.java

```diff
@@ -324,11 +324,22 @@ private static long getBinlogTimestamp(BinaryLogClient client, String binlogFile
             client.setBinlogFilename(binlogFile);
             client.setBinlogPosition(0);
 
-            LOG.info("begin parse binlog: {}", binlogFile);
+            LOG.info("Begin parse binlog: {}", binlogFile);
             client.connect();
         } finally {
             client.unregisterEventListener(eventListener);
         }
+        if (binlogTimestamps.isEmpty()) {
+            try {
+                if (client.isConnected()) {
```

Comment:
IMO this condition should always be false.

Reply:
> IMO this condition should always be false.

yes